src/wasm/wasm-stack.cpp

```diff
@@ -18,10 +18,20 @@
 
 namespace wasm {
 
+void BinaryInstWriter::emitBlockType(Type type) {
```

Comment:
the name suggests it's just for Blocks, how about `emitControlFlowType`? or maybe `emitGeneralType`? (but not sure that's a good name for "might be non-single")

Reply:
How about `emitResultType`?

Reply:
sgtm, better than the others for sure.